homeassistant/components/zha/binary_sensor.py

```diff
@@ -166,13 +145,26 @@ def async_set_state(self, state):
     async def async_update(self):
         """Attempt to retrieve on off state from the binary sensor."""
         await super().async_update()
-        if self._on_off_channel:
-            self._state = await self._on_off_channel.get_attribute_value("on_off")
-        if self._zone_channel:
-            value = await self._zone_channel.get_attribute_value("zone_status")
-            if value is not None:
-                self._state = value & 3
-        if self._attr_channel:
-            self._state = await self._attr_channel.get_attribute_value(
-                self._attr_channel.value_attribute
+        if hasattr(self._channel, "value_attribute"):
+            self._state = await self._channel.get_attribute_value(
+                self._channel.value_attribute
             )
+        else:
+            self._state = await self._channel.get_attribute_value("on_off")
+
+
+@STRICT_MATCH(MatchRule(channel_names={CHANNEL_ZONE}))
+class BinarySensor(BinarySensor):
```

Comment:
Shouldn’t we give these different names? I think python allows this because the bases are different but it’s confusing for anyone that has to maintain this IMO.

Reply:
Addressed in https://github.com/home-assistant/home-assistant/pull/30138/commits/e7a145328ca67d684d31598148a1ed9a9ea98fb6